assets/js/blocks/checkout/inner-blocks/checkout-billing-address-block/edit.tsx

```diff
@@ -41,12 +46,30 @@ export const Edit = ( {
 	} = useCheckoutBlockContext();
 	const { addressFieldControls: Controls } =
 		useCheckoutBlockControlsContext();
-	const { showBillingFields } = useCheckoutAddress();
+	const { showBillingFields, forcedBillingAddress } = useCheckoutAddress();
 
-	if ( ! showBillingFields ) {
+	if ( ! showBillingFields && ! forcedBillingAddress ) {
 		return null;
 	}
-
+	if ( forcedBillingAddress ) {
+		attributes.title =
+			attributes.title === DEFAULT_TITLE
```

Comment:
This ternary operator is a little hard to reason with, maybe adding comments would be useful here.

Reply:
I would also consider if we need the `if/else` structure here, or if it would be better to move this logic into a separate function, remove the use of ternary operators and make this easier to read, what do you think?

Reply:
That's actually a good idea! I have moved the logic to a different function and added the comments for better understanding. Thank you! 🙂